src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberMethodSymbol.cs

```diff
@@ -1220,30 +1220,15 @@ private void DecodeWellKnownAttributeAppliedToMethod(ref DecodeWellKnownAttribut
             else if (VerifyObsoleteAttributeAppliedToMethod(ref arguments, AttributeDescription.DeprecatedAttribute))
             {
             }
-            else if (attribute.IsTargetAttribute(this, AttributeDescription.IsReadOnlyAttribute))
+            else if (ReportExplicitUseOfReservedAttributes(ref arguments,
```

Comment:
>ReportExplicitUseOfReservedAttributes [](start = 21, length = 37)

Just a heads up, this will conflict with local function attributes changes, which moves these checks into the `SourceMethodSymbolWithAttributes type